src/d3a/d3a_core/global_objects.py

```diff
@@ -0,0 +1,39 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+
+class GlobalObjects:
+
+    def __init__(self):
+        self.area_tree_dict = {}
+
+    def update(self, area):
+        if area.current_market:
+            self._create_grid_tree_dict(area, self.area_tree_dict)
+
+    def _create_grid_tree_dict(self, area, outdict):
+        outdict[area.name] = {}
+        if area.children:
+            outdict[area.name].update(
+                {"last_market_slot": area.current_market.time_slot_str,
+                 "last_market_bill": area.stats.get_current_market_stats_for_grid_tree(),
+                 "last_market_stats": area.stats.get_price_stats_current_market(),
+                 "last_market_fee": str(area.current_market.fee_class.grid_fee_rate),
```

Comment:
Would be better to prefix all with `current_` rather than `last_`.
Thanks!

Reply:
@faizan2590 This was a specific request from Colin and I agree. From a clients perspective, it is really the last market slot. If you look into the implementation of the current market, it is really the last past market (`past_markets[-1]`)

Reply:
Okay, Im indifferent to naming being followed here, but would press on having a uniform naming convention, at least for future to not being confused like `get_current_market_stats_for_grid_tree ` to be renamed to `get_last_market_stats_for_grid_tree ` and `get_current_market_maker_rate `  to `get_last_market_maker_rate ` .

Reply:
I would agree in general, but in the code, we use `current_market` as a name for the last_past_market. So either we rename it in the whole code (which is a huge refactoring story) or we keep it as is. I would go with the latter.
Maybe @spyrostz has a better idea?

Reply:
Okay, but you could at least rename `get_current_market_stats_for_grid_tree `, the one you added in this PR.

Reply:
:-) Sorry Faizan, but to be consistent, I dont agree. The `get_current_market_stats_for_grid_tree` is indeed using the current_market.

Reply:
> :-) Sorry Faizan, but to be consistent, I dont agree. The get_current_market_stats_for_grid_tree is indeed using the current_market.

You are adding the value of `current_market ` being returned via `get_current_market_stats_for_grid_tree` to `last_market_bill `.
Since this is just added in this PR, you can simply rename `get_current_market_stats_for_grid_tree` to `get_last_market_stats_for_grid_tree`

Reply:
OK, i give up, really not worth this long discussion here. Renaming it.